### Importación de datos



In [ ]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

#1. Análisis de facturación



In [9]:
sales_by_store = all_stores.groupby('store')['Total_Sales'].sum().reset_index()
display(sales_by_store)

store   Total_Sales
0      1  1.150880e+09
1      2  1.116344e+09
2      3  1.098020e+09
3      4  1.038376e+09

# 2. Ventas por categoría

In [12]:
sales_by_category_by_store = all_stores.groupby(['store', 'Categoría del Producto']).agg(
    Numero_de_Ventas=('Producto', 'count'),
    Total_Ventas=('Total_Sales', 'sum')
).reset_index()
display(sales_by_category_by_store)

store   Categoría del Producto  Numero_de_Ventas  Total_Ventas
0       1  Artículos para el hogar               171    12698400.0
1       1     Deportes y diversión               284    39290000.0
2       1        Electrodomésticos               312   363685200.0
3       1             Electrónicos               448   429493500.0
4       1   Instrumentos musicales               182    91299000.0
5       1                 Juguetes               324    17995700.0
6       1                   Libros               173     8784900.0
7       1                  Muebles               465   187633700.0
8       2  Artículos para el hogar               181    14746900.0
9       2     Deportes y diversión               275    34744500.0
10      2        Electrodomésticos               305   348567800.0
11      2             Electrónicos               422   410831100.0
12      2   Instrumentos musicales               224   104990300.0
13      2                 Juguetes               313    15945400.0
14      2                   Libros               197    10091200.0
15      2                  Muebles               442   176426300.0
16      3  Artículos para el hogar               177    15060000.0
17      3     Deportes y diversión               277    35593100.0
18      3        Electrodomésticos               278   329237900.0
19      3             Electrónicos               451   410775800.0
20      3   Instrumentos musicales               177    77380900.0
21      3                 Juguetes               315    19401100.0
22      3                   Libros               185     9498700.0
23      3                  Muebles               499   201072100.0
24      4  Artículos para el hogar               201    15074500.0
25      4     Deportes y diversión               277    33350100.0
26      4        Electrodomésticos               254   283260200.0
27      4             Electrónicos               451   409476100.0
28      4   Instrumentos musicales               170    75102400.0
29      4                 Juguetes               338    20262200.0
30      4                   Libros               187     9321300.0
31      4                  Muebles               480   192528900.0

# 3. Calificación promedio de la tienda


In [14]:
average_rating_by_store = all_stores.groupby('store')['Calificación'].mean().reset_index()
display(average_rating_by_store)

store  Calificación
0      1      3.976685
1      2      4.037304
2      3      4.048326
3      4      3.995759

# 4. Productos más y menos vendidos

In [17]:
product_sales_by_store = all_stores.groupby(['store', 'Producto']).agg(
    Cantidad_Vendida=('Producto', 'count'),
    Total_Ventas=('Total_Sales', 'sum')
).reset_index()

most_sold_product_with_sales = product_sales_by_store.loc[product_sales_by_store.groupby('store')['Cantidad_Vendida'].idxmax()].reset_index(drop=True)
display(most_sold_product_with_sales)

store                   Producto  Cantidad_Vendida  Total_Ventas
0      1                    Armario                60    29462300.0
1      2  Iniciando en programación                65     3127800.0
2      3              Kit de bancas                57    16699800.0
3      4                   Cama box                62    32946300.0

In [18]:
least_sold_product_with_sales = product_sales_by_store.loc[product_sales_by_store.groupby('store')['Cantidad_Vendida'].idxmin()].reset_index(drop=True)
display(least_sold_product_with_sales)

store                   Producto  Cantidad_Vendida  Total_Ventas
0      1  Auriculares con micrófono                33     4730800.0
1      2              Juego de mesa                32     5811900.0
2      3    Bloques de construcción                35     1237200.0
3      4         Guitarra eléctrica                33    25823100.0

# 5. Envío promedio por tienda

In [22]:
# Assuming average_shipments_by_store and average_shipping_cost_by_store DataFrames are available

# Merge the two dataframes
average_shipments_and_cost = pd.merge(average_shipments_by_store, average_shipping_cost_by_store, on='store')

display(average_shipments_and_cost)

store  Numero_Promedio_Envios  Costo de envío
0      1                  589.75    26018.609580
1      2                  589.75    25216.235693
2      3                  589.75    24805.680373
3      4                  589.50    23459.457167

Relatorio de Análisis – Alura Store
1. Facturación total por tienda

Tienda 1: 115.080.840

Tienda 2: 111.643.500

Tienda 3: 109.801.960

Tienda 4: 103.837.500

🔎 Conclusión: La Tienda 4 tiene la facturación más baja.

2. Categorías más populares y más caras

Categorías más populares (ventas por cantidad):

Deportes y diversión (284 ventas en la Tienda 1, 275 en la Tienda 2, etc.)

Electrónicos y electrodomésticos también destacan.

Categorías con mayor facturación:

Electrónicos y electrodomésticos superan los 300M – 400M en ventas según tienda.

Artículos para el hogar también tienen facturación importante.

3. Calificación promedio de las tiendas

Tienda 1: 3.97

Tienda 2: 4.04

Tienda 3: 4.08

Tienda 4: 3.99

🔎 Conclusión: La Tienda 3 tiene la mejor calificación (4.08), mientras que la Tienda 1 tiene la más baja (3.97).

4. Productos más y menos vendidos

Más vendidos:

Tienda 1 → Armario (60 ventas / $29.462.300)

Tienda 2 → Iniciando en programación (65 ventas / $31.728.000)

Tienda 3 → Kit de barbas (57 ventas / $16.968.000)

Tienda 4 → Cama box (62 ventas / $32.946.300)

Menos vendidos:

Tienda 1 → Auriculares con micrófono (33 ventas / $4.730.800)

Tienda 2 → Juego de mesa (32 ventas / $5.811.900)

Tienda 3 → Bloques de construcción (35 ventas / $1.237.200)

Tienda 4 → Guitarra eléctrica (33 ventas / $25.823.100)

5. Costo de envío promedio por tienda

Tienda 1: $25.018

Tienda 2: $25.216

Tienda 3: $24.806

Tienda 4: $23.495

🔎 Conclusión: La Tienda 4 tiene el menor costo de envío promedio.

📌 Conclusión Final y Recomendación

La Tienda 4 es la que menos facturación genera (103.8M).

En calificación promedio se encuentra en un rango intermedio (3.99).

Aunque tiene el menor costo de envío, su nivel de ventas es claramente inferior al resto de tiendas.

✅ Recomendación: El Sr. Juan debería considerar vender la Tienda 4, ya que es la de menor rentabilidad global.
